li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaInstrumentedProducerImpl.java

```diff
@@ -65,6 +71,10 @@
   private volatile Producer<K, V> delegate;
   private final SimpleClient mdsClient;
 
+  // This is null if the underlying producer does not have an implementation for time-bounded flush
+  private Method boundedFlushMethod;
```

Comment:
final ?

Reply:
This is in recreateDelegate(), so declaring it final gives a "cannot assign" error.